LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs

```diff
@@ -132,14 +133,34 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                    {
                        _ = endpoints.MapGet(BasicsStationNetworkServer.DiscoveryEndpoint, async context =>
                            {
+                               if (context.Request.Host.Port is not (BasicsStationNetworkServer.LnsPort or BasicsStationNetworkServer.LnsSecurePort))
+                               {
+                                   context.Response.StatusCode = (int)System.Net.HttpStatusCode.MethodNotAllowed;
+                                   return;
+                               }
                                var lnsProtocolMessageProcessor = context.RequestServices.GetRequiredService<ILnsProtocolMessageProcessor>();
                                await lnsProtocolMessageProcessor.HandleDiscoveryAsync(context, context.RequestAborted);
                            });
                        _ = endpoints.MapGet($"{BasicsStationNetworkServer.DataEndpoint}/{{{BasicsStationNetworkServer.RouterIdPathParameterName}:required}}", async context =>
                            {
+                               if (context.Request.Host.Port is not (BasicsStationNetworkServer.LnsPort or BasicsStationNetworkServer.LnsSecurePort))
+                               {
+                                   context.Response.StatusCode = (int)System.Net.HttpStatusCode.MethodNotAllowed;
+                                   return;
+                               }
                                var lnsProtocolMessageProcessor = context.RequestServices.GetRequiredService<ILnsProtocolMessageProcessor>();
                                await lnsProtocolMessageProcessor.HandleDataAsync(context, context.RequestAborted);
                            });
+                       _ = endpoints.MapPost(BasicsStationNetworkServer.UpdateInfoEndpoint, async context =>
```

Comment:
I think a lot of the duplication between mapping the handler, filtering for allowed conditions and invoking the handler can be consolidated via a local function:

```c#
Map(HttpMethod.Get, BasicsStationNetworkServer.DiscoveryEndpoint,
    context => context.Request.Host.Port is BasicsStationNetworkServer.LnsPort or BasicsStationNetworkServer.LnsSecurePort,
    (ILnsProtocolMessageProcessor processor) => processor.HandleDiscoveryAsync);

Map(HttpMethod.Get, $"{BasicsStationNetworkServer.DataEndpoint}/{{{BasicsStationNetworkServer.RouterIdPathParameterName}:required}}",
    context => context.Request.Host.Port is BasicsStationNetworkServer.LnsPort or BasicsStationNetworkServer.LnsSecurePort,
    (ILnsProtocolMessageProcessor processor) => processor.HandleDataAsync);

Map(HttpMethod.Post, BasicsStationNetworkServer.UpdateInfoEndpoint,
    context => context.Connection.LocalPort is BasicsStationNetworkServer.CupsPort,
    (ICupsProtocolMessageProcessor processor) => processor.HandleUpdateInfoAsync);

void Map<TService>(HttpMethod method, string pattern,
                   Predicate<HttpContext> predicate,
                   Func<TService, Func<HttpContext, CancellationToken, Task>> handlerMapper)
{
    _ = endpoints.MapMethods(pattern, new[] { method.ToString() }, async context =>
    {
        if (!predicate(context))
        {
            context.Response.StatusCode = (int)System.Net.HttpStatusCode.MethodNotAllowed;
            return;
        }
        var processor = context.RequestServices.GetRequiredService<TService>();
        var handler = handlerMapper(processor);
        await handler(context, context.RequestAborted);
    });
}
```

Note that with the predicate being inverted, the conditions are simpler and specify the positive case.

Reply:
Thanks for the suggestion! I will bring this in